ligo/stablecoin/actions.ligo

```diff
@@ -240,4 +339,230 @@ function is_operator_action
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+
+; const senderAddress : address = Tezos.sender
+
+; function mint_tokens
+    ( const accumulator       : storage
+    ; const mint_param        : mint_param
+    ; const current_allowance : nat
+    ) : storage is block
+  { const unwrapped_parameter : mint_param_ =
+      Layout.convert_from_right_comb ((mint_param : mint_param))
+  ; fail_on
+      ( unwrapped_parameter.amount > current_allowance
+      , "Requested amount of minting tokens exceeds current allowance"
+      )
+
+  ; const updated_allowance : nat =
+      abs (current_allowance - unwrapped_parameter.amount)
+  ; const updated_allowances : minting_allowances =
+      Big_map.update (Tezos.sender, Some (updated_allowance), accumulator.minting_allowances)
+  ; const updated_store : storage = accumulator with record
+      [ minting_allowances = updated_allowances ]
+  } with credit_to (unwrapped_parameter, updated_store)
+
+} with
+  ( (nil : list (operation))
+  , List.fold
+      ( function
+          ( const accumulator : storage
+          ; const mint_param  : mint_param
+          ) : storage is case accumulator.minting_allowances[senderAddress] of
+            None -> (failwith ("Sender address is not minter") : storage)
+          | Some (current_allowance) ->
+              mint_tokens (accumulator, mint_param, current_allowance)
+          end
+      , parameters
+      , store
+      )
+  )
+
+(*
+ * Decreases balance and total supply of otkens by the given amount
+ *)
+function burn
+  ( const parameters : burn_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+; const sender_address : address = Tezos.sender
+} with
+  ( (nil : list (operation))
+  , List.fold
+      ( function
+          ( const accumulator : storage
+          ; const burn_amount : nat
+          ) : storage is debit_from
+            ( record
+                [ from_ = sender_address
+                ; amount = burn_amount
+                ]
+            , accumulator
+            )
+      , parameters
+      , store
+      )
+  )
+
+(*
+ * Sets token owner to a new address
+ *)
+function transfer_ownership
+  ( const parameter : transfer_ownership_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_owner (store)
+} with
+  ( (nil : list (operation))
+  , store with record [roles.pending_owner = Some (parameter)]
+  )
+
+(*
+ * When pending sender calls this entrypoint, the address
+ * receives owner privileges for contract
+ *)
+function accept_ownership
+  ( const parameter : accept_ownership_param
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_pending_owner (store)
+} with
+  ( (nil : list (operation))
+  , store with record
+      [ roles.pending_owner = (None : option (address))
+      // We have already validated pending owner
+      ; roles.owner = Tezos.sender
```

Comment:
It might be better to be defensive and explicitly use the value in the `pending_owner` field here, instead of using the sender value.

Reply:
I wonder if it requires more instructions

Reply:
On second look, I fear this appears kind of important. Because if by some accident, the `authorize_pending_owner` call is removed or if there is a bug in it (a bad rebase, for example), it will give any random sender owner privileges. @gromakovsky What do you think?

I think `authorize_pending_owner` can be made to return the address of pending owner as `address` (unwrapping it from `Some` value or failing otherwise), and we can just use it here. Shouldn't add a lot more instructions, right?

Reply:
> I think authorize_pending_owner can be made to return the address of pending owner as `address`

Okay, sounds cheap indeed, let's do it.